src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -1053,12 +1053,17 @@ void Terminal::_AdjustCursorPosition(const COORD proposedPosition)
 
     // Firing the CursorPositionChanged event is very expensive so we try not to do that when
     // the cursor does not need to be redrawn.
-    if (!cursor.IsDeferDrawing())
+    if (!cursor.IsDeferDrawing() || _trackingCursorMovement)
     {
         _NotifyTerminalCursorPositionChanged();
     }
 }
 
+void Terminal::TrackCursorMovement(bool track) noexcept
+{
+    _trackingCursorMovement = track;
+}
+

```

Comment:
I am not sure how I feel about this - it feels like this is not something the terminal needs to know. An alternative I considered was having `ControlCore` tell `TermControl` that the cursor moved in `ControlCore::_connectionOutputHandler`, but it is not necessarily the case that the cursor always moves when there's output from the connection.